### ISSUE TITLE
[Feature request] Remove Markdown Characters from Content Preview

### ISSUE THREAD

**Issue by PaulWen** [scambier/obsidian-omnisearch#86]
**Is your feature request related to a problem? Please describe.**

I feel like the markdown characters shown in the content preview are distracting me from the actual content and also take up valuable space for further search results or more context.

**Describe the solution you'd like**

I would love to have the option to see the content preview without any markdown characters (e.g. `[[...]]`, `#`, `##`, `==...==`, `~~...~~`).

**Comment by scambier** [comment]
Been there, done that, [and reverted it](https://github.com/scambier/obsidian-omnisearch/commit/e949f0916aa2a02ade0d0e5ef4b93735f44a047a). I can remember several issues that I tried to fix before giving up:

- Stripping markdown chars is not as straightforward as it seems, and the "best" way is to render the html. Unfortunately that takes time, and can cause security issues when not done properly.
- Some libraries can take care of this, but they're just as slow and make Omnisearch way heavier. Same problem with Obsidian's native renderer.
- A lot of users have plugins that add custom markdown syntax, and you might inadvertently remove searchable information.
- It complexifies indexing, highlighting and cursor positionning, because of the discrepancies between raw and stripped markdown.

All in all, I deemed it not worth it 🤷‍♂️
